File: algorithms/easy/9_palindrome_number.py
# 9. Palindrome Number

def is_palindrome(x):
    """Determine if an integer is a palindrome.
    
    A palindrome reads the same forwards and backwards.
    
    Args:
        x: Integer to check
    
    Returns:
        Boolean indicating if x is a palindrome
    """
    # Negative numbers are not palindromes
    if x < 0:
        return False
    
    # Convert to string and check if it equals its reverse
    return str(x) == str(x)[::-1]

# Alternative solution without converting to string
def is_palindrome_math(x):
    if x < 0 or (x % 10 == 0 and x != 0):
        return False
    
    reversed_num = 0
    while x > reversed_num:
        reversed_num = reversed_num * 10 + x % 10
        x //= 10
    
    # When length is odd, we need to remove the middle digit
    return x == reversed_num or x == reversed_num // 10

# Test cases
print(is_palindrome(121))   # True
print(is_palindrome(-121))  # False
print(is_palindrome(10))    # False

File: algorithms/easy/9_palindrome_number.js
/**
 * 9. Palindrome Number
 * 
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    // Negative numbers are not palindromes
    if (x < 0) return false;
    
    // Convert to string and check if it equals its reverse
    const str = x.toString();
    return str === str.split('').reverse().join('');
};

// Alternative solution without converting to string
var isPalindromeMath = function(x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) return false;
    
    let reversed = 0;
    while (x > reversed) {
        reversed = reversed * 10 + x % 10;
        x = Math.floor(x / 10);
    }
    
    // When length is odd, we need to remove the middle digit
    return x === reversed || x === Math.floor(reversed / 10);
};

// Test cases
console.log(isPalindrome(121));   // true
console.log(isPalindrome(-121));  // false
console.log(isPalindrome(10));    // false